cmd/internal/cli/pull.go

```diff
@@ -62,10 +64,20 @@ var pullArchFlag = cmdline.Flag{
 	Value:        &pullArch,
 	DefaultValue: runtime.GOARCH,
 	Name:         "arch",
-	Usage:        "architecture to pull from library",
+	Usage:        "architecture for images to pull from library",
 	EnvKeys:      []string{"PULL_ARCH"},
 }
 
+// --platform
+var pullPlatformFlag = cmdline.Flag{
+	ID:           "pullPlatformFlag",
+	Value:        &pullPlatform,
+	DefaultValue: "",
+	Name:         "platform",
+	Usage:        "platform for images to pull from OCI sources",
```

Comment:
On the UX front, I wonder if this flag could/should work with `library` sources as well, potentially deprecating the `arch` flag?

Reply:
Opened - https://github.com/sylabs/singularity/issues/2015